syntax = "proto3";
package query_service_ts;

service QueryServiceTs {
  rpc QueryBySchema(SchemaId) returns (TimeSeries);
  rpc QueryByRange(Range) returns (TimeSeries);
}

service QueryServiceTsRaw {
  rpc QueryRaw (RawStatement) returns (ValueBytes);
}

message Range {
  string schema_id = 1;
  string object_id = 2;
  string start = 3;
  string end = 4;
  string step = 5;
}

message SchemaId {
  string schema_id = 1;
}

message TimeSeries {
  string timeseries = 1;
}

message RawStatement {
  string raw_statement = 1;
}

message ValueBytes {
  bytes value_bytes = 1;
}
